src/ocamldep.ml

```diff
@@ -42,6 +45,32 @@ module Dep_graph = struct
         (Fmt.list ~pp_sep:Fmt.nl (Fmt.prefix (Fmt.string "-> ") Module.Name.pp))
         (List.map cycle ~f:Module.name)
 
+  module Multi = struct
+    let top_closed_multi (ts : t list) modules =
+      List.concat_map ts ~f:(fun t ->
+        Module.Name.Map.to_list t.per_module
+        |> List.map ~f:(fun (_name, (unit, deps)) ->
+          deps >>^ fun deps -> (unit, deps)))
+      |> Build.all >>^ fun per_module ->
+      let per_obj =
+        Module.Obj_map.of_list_reduce per_module ~f:List.rev_append in
+      match Module.Obj_map.top_closure per_obj modules with
```

Comment:
We should be able to factorise this code with the one in `top_closed`

Reply:
The errors are actually a bit different as one contains a directory. I've factorised the pretty printing of the cycle and the other function though.